### ISSUE TITLE
v1.16 Add filter for search titles

### ISSUE THREAD

**Issue by felixhahnweilheim** [humhub/humhub#6947]
### Is your feature request related to a problem? Please describe.
The Verified module adds an icon after the name of a verified user/space.
The new search dropdown does not allow to add this icon. There are no widgets to overwrite or other options to properly add the icon.

### Describe the solution you'd like
Add an event to `getTitle()` in https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/user/search/SearchRecord.php#L48 (and same for https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/space/search/SearchRecord.php)
that allows to manipulate the title. In our usecase it would even be enough to add something after the title.

I did not test that yet. I first wanted to ask if you would be open to such a PR or what your thoughts are.

### Describe alternatives you've considered
If you see alternatvies, please let me know.

**Comment by luke-** [comment]
If we extend `UserSearchResult` from `Object` and use `Yii::configure()` here,
https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/user/search/UserSearchProvider.php#L72

the module could replace the `SearchResult` class, extend it and implement an own `getTitle()` method.

@yurabakhtin @felixhahnweilheim  What do you think? 


```
    'container' => [
        'definitions' => [
            \humhub\modules\user\search\SearchRecord::class => ['class' => \custom\SearchRecord]
        ]
    ] 
```

**Comment by felixhahnweilheim** [comment]
That sounds good. @ArchBlood , do you agree?

**Comment by ArchBlood** [comment]
> That sounds good. @ArchBlood , do you agree?

Would make things easier, this would also add other possibilities as well.

**Comment by ArchBlood** [comment]
> If we extend `UserSearchResult` from `Object` and use `Yii::configure()` here, https://github.com/humhub/humhub/blob/develop/protected/humhub/modules/user/search/UserSearchProvider.php#L72
> 
> the module could replace the `SearchResult` class, extend it and implement an own `getTitle()` method.
> 
> @yurabakhtin @felixhahnweilheim What do you think?
> 
> ```
>     'container' => [
>         'definitions' => [
>             \humhub\modules\user\search\SearchRecord::class => ['class' => \custom\SearchRecord]
>         ]
>     ] 
> ```

Looking into this more I'd like to note some possible downsides such as increased complexity and maintainability from extending the core classes, debugging could also be affected here if the developer isn't careful, if other modules also inject into the core functions this could cause unknown issues between modules if developers aren't careful, currently with how the Verified module is implemented we inject the [VerifiedIcon](https://github.com/GreenMeteor/verified/blob/main/widgets/VerifiedIcon.php) widget into a number of containers through slightly hackish ways through our [VerifiedService](https://github.com/GreenMeteor/verified/blob/main/services/VerifiedService.php) class using [yii\base\WidgetEvent](https://github.com/GreenMeteor/verified/blob/main/Events.php) & [EVENT_BEFORE_RUN](https://github.com/GreenMeteor/verified/blob/main/config.php), even more so when it comes to the People & Spaces sections. If @yurabakhtin needs access to view the linked code I'm more than willing to allow this as well.

**Comment by yurabakhtin** [comment]
@luke- 

> What do you think?
> ```
>     'container' => [
>         'definitions' => [
>             \humhub\modules\user\search\SearchRecord::class => ['class' => \custom\SearchRecord]
>         ]
>     ] 
> ```

I have tested this and it doesn't work because we call the class as `new SearchRecord($user)`.
It works only if call it as `Yii::createObject(SearchRecord::class, [$user])`.
Do you agree to make these changes for at least core search providers?

---

@ArchBlood @felixhahnweilheim After we will do the changes above you can handle the class by event from your moduke side like this:

- `config.php`:

```php
[Application::class, Application::EVENT_ON_INIT, [Events::class, 'onApplicationInit']],
```

- `Events.php`:

```php
    public static function onApplicationInit($event)
    {
        /* @var Application $app */
        $app = $event->sender;
        $app->setContainer(['definitions' => [
            user\SearchRecord::class => your_module\VerifiedUserSearchRecord::class,
        ]]);
    }
```

- `your_module/VerifiedUserSearchRecord.php`

```php
<?php
namespace humhub\modules\your_module\search;

use humhub\modules\ui\icon\widgets\Icon;
use humhub\modules\user\search\SearchRecord;

class VerifiedUserSearchRecord extends SearchRecord
{
    /**
     * @inheritdoc
     */
    public function getTitle(): string
    {
        return parent::getTitle() . Icon::get('check');
    }
}
```

**Comment by ArchBlood** [comment]
> @luke-
> 
> > What do you think?
> > ```
> >     'container' => [
> >         'definitions' => [
> >             \humhub\modules\user\search\SearchRecord::class => ['class' => \custom\SearchRecord]
> >         ]
> >     ] 
> > ```
> 
> I have tested this and it doesn't work because we call the class as `new SearchRecord($user)`. It works only if call it as `Yii::createObject(SearchRecord::class, [$user])`. Do you agree to make these changes for at least core search providers?
> 
> @ArchBlood @felixhahnweilheim After we will do the changes above you can handle the class by event from your moduke side like this:
> 
> * `config.php`:
> 
> ```
> [Application::class, Application::EVENT_ON_INIT, [Events::class, 'onApplicationInit']],
> ```
> 
> * `Events.php`:
> 
> ```
>     public static function onApplicationInit($event)
>     {
>         /* @var Application $app */
>         $app = $event->sender;
>         $app->setContainer(['definitions' => [
>             user\SearchRecord::class => your_module\VerifiedUserSearchRecord::class,
>         ]]);
>     }
> ```
> 
> * `your_module/VerifiedUserSearchRecord.php`
> 
> ```
> <?php
> namespace humhub\modules\your_module\search;
> 
> use humhub\modules\ui\icon\widgets\Icon;
> use humhub\modules\user\search\SearchRecord;
> 
> class VerifiedUserSearchRecord extends SearchRecord
> {
>     /**
>      * @inheritdoc
>      */
>     public function getTitle(): string
>     {
>         return parent::getTitle() . Icon::get('check');
>     }
> }
> ```

I like the approach here, it wouldn't take as much time and maintainability would be consistent with the core.

**Comment by luke-** [comment]
@yurabakhtin Thanks for pointing out!

> I have tested this and it doesn't work because we call the class as `new SearchRecord($user)`. It works only if call it as `Yii::createObject(SearchRecord::class, [$user])`. Do you agree to make these changes for at least core search providers?
> 

Yes, let's do it that way!

**Comment by yurabakhtin** [comment]
@luke- Done in the PR https://github.com/humhub/humhub/pull/7052.

**Comment by ArchBlood** [comment]
> @luke-
> 
> > What do you think?
> > ```
> >     'container' => [
> >         'definitions' => [
> >             \humhub\modules\user\search\SearchRecord::class => ['class' => \custom\SearchRecord]
> >         ]
> >     ] 
> > ```
> 
> I have tested this and it doesn't work because we call the class as `new SearchRecord($user)`. It works only if call it as `Yii::createObject(SearchRecord::class, [$user])`. Do you agree to make these changes for at least core search providers?
> 
> @ArchBlood @felixhahnweilheim After we will do the changes above you can handle the class by event from your moduke side like this:
> 
> * `config.php`:
> 
> ```
> [Application::class, Application::EVENT_ON_INIT, [Events::class, 'onApplicationInit']],
> ```
> 
> * `Events.php`:
> 
> ```
>     public static function onApplicationInit($event)
>     {
>         /* @var Application $app */
>         $app = $event->sender;
>         $app->setContainer(['definitions' => [
>             user\SearchRecord::class => your_module\VerifiedUserSearchRecord::class,
>         ]]);
>     }
> ```
> 
> * `your_module/VerifiedUserSearchRecord.php`
> 
> ```
> <?php
> namespace humhub\modules\your_module\search;
> 
> use humhub\modules\ui\icon\widgets\Icon;
> use humhub\modules\user\search\SearchRecord;
> 
> class VerifiedUserSearchRecord extends SearchRecord
> {
>     /**
>      * @inheritdoc
>      */
>     public function getTitle(): string
>     {
>         return parent::getTitle() . Icon::get('check');
>     }
> }
> ```

@luke- @yurabakhtin After finally having the time to attempt to implement this method `getTitle()` is being sanitized making this method impossible for our use case.

### What happens when we use this method
![Screenshot_1](https://github.com/user-attachments/assets/85913943-d7ca-4eb0-a464-d4084a988acd)

### Expected Result
![Screenshot_2](https://github.com/user-attachments/assets/27992bbb-2a0f-4091-868d-4afbc24dd6a4)

**Comment by yurabakhtin** [comment]
@ArchBlood Yes, you are right, the html was removed by JS side.
@luke- Fixed in PR https://github.com/humhub/humhub/pull/7278. I have removed the JS code because in previous version a post content was rendered by JS side with widget `ui.richtext.prosemirror.RichText`, but recently we implemented the rendering on PHP side [here](https://github.com/humhub/humhub/pull/7193/files#diff-c4cb0fcf397d1272c6dd36f78b88a309cf38639b6947ca4258ca8eccd8d25e93R49) with `strip_tags()`, so we don't need the rendering by JS `ui.richtext.prosemirror.RichText` now and no need to remove html tags on JS side as well.
